compiler/rustc_mir/src/const_eval/machine.rs

```diff
@@ -231,8 +232,13 @@ impl<'mir, 'tcx> interpret::Machine<'mir, 'tcx> for CompileTimeInterpreter<'mir,
             if ecx.tcx.is_const_fn_raw(def.did) {
                 // If this function is a `const fn` then under certain circumstances we
                 // can evaluate call via the query system, thus memoizing all future calls.
-                if ecx.try_eval_const_fn_call(instance, ret, args)? {
-                    return Ok(None);
+                match instance.def {
+                    InstanceDef::Intrinsic(_) => {
```

Comment:
ah sorry, I meant to run this check within `try_eval_const_fn_call` and return `Ok(false)` from it if it's not an intrinsic

Reply:
I don't understand this new code. Don't we have to just entirely remove memoization?

Reply:
we still memoize calls to `size_of` and `align_of` I think

Reply:
Is that worth it all the code complexity here? Those functions are trivial to evaluate...